src/core/notmuch_data_source.py

```
@@ -39,12 +41,16 @@
     Enhanced data source for Notmuch with AI analysis and tagging support.
-    
+
     This implementation provides both read and write capabilities for Notmuch,
     along with AI-powered analysis and smart filtering.
     """
 
-    def __init__(self, db_path: Optional[str] = None, db_manager: Optional['DatabaseManager'] = None):
+    def __init__(
+        self,
+        db_path: Optional[str] = None,
+        db_manager: Optional["DatabaseManager"] = None,
```

Flake8 (7.3.0)

[error] undefined name 'DatabaseManager'

(F821)